### ISSUE TITLE
Enhancement request: lift requirement for parenthetical syntax when using pipes

### ISSUE THREAD

**Issue by brandondrew** [elixir-lang/elixir#3181]
In `iex` I can use parentheses to nest an expression as the argument for `Enum.count`, but I can't pipe the results of an expression to `Enum.count`

~~~
iex(23)> Enum.count( Enum.concat [1..3] )
3
iex(24)> Enum.concat [1..3] |> Enum.count
** (Protocol.UndefinedError) protocol Enumerable not implemented for 1
    (elixir) lib/enum.ex:1: Enumerable.impl_for!/1
    (elixir) lib/enum.ex:112: Enumerable.reduce/3
    (elixir) lib/enum.ex:1274: Enum.reduce/3
    (elixir) lib/enum.ex:416: Enum.do_concat/1
~~~

However I can work around this by putting parentheses around the arguments of the first expression

    Enum.concat([1..3]) |> Enum.count

Would it be possible to allow the simpler version (`Enum.concat [1..3] |> Enum.count`) to work?  Two of Elixir's very nice syntactic features are its very readable piping-style syntax, and its optional parentheses.   It would be a great shame if these two features could not be used simultaneously—if you had to choose one or the other.  The code is much more readable (for humans, at least) with fewer jots and tittles.  I realize it will be more convenient to parse or compile it with parentheses, but is it *possible* to make Elixir work without requiring the parentheses?

**Comment by chrismccord** [comment]
Elixir's precedence rules make the version without parens different than you intend: 

```elixir
iex(2)> (quote do: Enum.concat [1..3] |> Enum.count) |> Macro.to_string
"Enum.concat([1 .. 3] |> Enum.count())"
iex(3)>
```
So parens are required to disambiguate the expression. In coming versions, the compiler will warn or error, exactly to prevent these issue. For feature requests/discussions, we have the elixir-lang-core mailing list https://groups.google.com/forum/#!forum/elixir-lang-core

**Comment by brandondrew** [comment]
Sorry for leaving this in the wrong place.  When I composed most of that it was (in my mind) a bug report, and only at the last minute I realized it could be a problem with actual precedence not matching my expected precedence...  I should have stopped and rethought the situation and checked for the appropriate place to file feature requests but I was getting called away to something else and didn't think through it clearly.  I'll try to keep this in mind in the future.
